### ISSUE TITLE
_new__() missing 2 required positional arguments: 'required' and 'optional'

### ISSUE THREAD

**Issue by li35619081** [python-gitlab/python-gitlab#1597]
## Description of the problem, including code/CLI snippet
pip install python-gitlab
python-gitlab 2.10.1 

, line 276, in RESTManager
    _create_attrs: RequiredOptional = RequiredOptional()
TypeError: __new__() missing 2 required positional arguments: 'required' and 'optional'
python-BaseException

## Expected Behavior
import gitlab

git = gitlab.Gitlab(url="our_gitlab_host", private_token="mytoken",per_page=1)

## Actual Behavior
when i use gitlab.Gitlab init, show error info.

## Specifications

  - python-gitlab version:  2.10.1 
  - API version you are using (v3/v4):  v4
  - Gitlab server version (or gitlab.com):

**Comment by JohnVillalovos** [comment]
I can't reproduce this. I tried with Python 3.9. Which version of Python are you using?

**Comment by li35619081** [comment]
> I can't reproduce this. I tried with Python 3.9. Which version of Python are you using?

I use with Python 3.6.

**Comment by JohnVillalovos** [comment]
I am unable to reproduce this with Python 3.6.

**Comment by mspbgx** [comment]
I got the same error. I use Python 3.6.0 with python-gitlab 2.10.1.

**Comment by nejch** [comment]
Can you paste the exact snippet and traceback (with redacted tokens)?

**Comment by mspbgx** [comment]
@nejch of course.
![image](https://user-images.githubusercontent.com/11456343/132806151-b79b8805-fe01-474a-8cdc-f4910004714b.png)

**Comment by nejch** [comment]
Thanks!

@JohnVillalovos I can reproduce this on python 3.6.0 but not later ones, for example, 3.6.15. Might be namedtuple bug in 3.6.0? Would need to investigate. @mspbgx @li35619081 in the meantime if you can slightly bump your python patch version, this should already fix it for your.

3.6.15
```
docker run --rm -it --entrypoint bash python:3.6
root@a1afeefede86:/# pip3 install --quiet python-gitlab
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is
recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
root@a1afeefede86:/# python3
Python 3.6.15 (default, Sep  8 2021, 02:11:58)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gitlab
>>> gl = gitlab.Gitlab("https://gitlab.com")
>>>

```

3.6.0
```
docker run --rm -it --entrypoint bash python:3.6.0
root@2505d41eaf8d:/# pip3 install --quiet python-gitlab
root@2505d41eaf8d:/# python3
Python 3.6.0 (default, Feb 28 2017, 22:20:46)
[GCC 4.9.2] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gitlab
>>> gl = gitlab.Gitlab("https://gitlab.com")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/gitlab/client.py", line 107, in __init__
    import gitlab.v4.objects
  File "/usr/local/lib/python3.6/site-packages/gitlab/v4/objects/__init__.py", line 18, in <module>
    from .access_requests import *
  File "/usr/local/lib/python3.6/site-packages/gitlab/v4/objects/access_requests.py", line 1, in <module>
    from gitlab.base import RESTManager, RESTObject
  File "/usr/local/lib/python3.6/site-packages/gitlab/base.py", line 267, in <module>
    class RESTManager(object):
  File "/usr/local/lib/python3.6/site-packages/gitlab/base.py", line 276, in RESTManager
    _create_attrs: RequiredOptional = RequiredOptional()
TypeError: __new__() missing 2 required positional arguments: 'required' and 'optional'
```

**Comment by nejch** [comment]
Specifically I think this was fixed in python 3.6.6, and might be a typing issue?

> bpo-28556: Minor fixes in typing module: add annotations to NamedTuple.__new__, pass *args and **kwds in Generic.__new__. Original PRs by Paulius Šarka and Chad Dombrova.

https://docs.python.org/release/3.6.12/whatsnew/changelog.html#python-3-6-6-final
https://bugs.python.org/issue28556

**Comment by mspbgx** [comment]
@nejch With Python 3.6.15 works it! Thanks!

**Comment by JohnVillalovos** [comment]
> Specifically this was fixed in python 3.6.1 (confirmed).

Great job root-causing this!

**Comment by JohnVillalovos** [comment]
Closing as seems to be resolved. Plus we will be deprecating Python 3.6 soon.
